openqa-label-known-issues

```diff
@@ -134,14 +134,15 @@ investigate_issue() {
     # against even in case when autoinst-log.txt is missing the details, e.g.
     # see https://progress.opensuse.org/issues/69178
     echo "$reason" >> "$out"
-    if [[ "$curl_output" != "200" ]] && [[ "$curl_output" != "301" ]]; then
+    # Control flow when there is a curl issue but still want to process job due to reason var
+    # even if autoinst-log.txt not found but assume is missing.
+    [[ -n "$reason" ]] && [[ "$curl_output" = "404" ]] && [[ "$reason" != null ]] \
+        && is_unreachachle_but_has_reason=1
+    if [[ "$curl_output" != "200" ]] && [[ "$curl_output" != "301" ]] && [[ -z "$is_unreachachle_but_has_reason" ]]; then
```

Comment:
With this change jobs that have a reason but no autoinst-log.txt are not handled in this branch anymore and checked if the job is maybe too old.
Is that the intention?

Also I find that long variable not very readable.
Also why not just use `true` and `false` to avoid having to think about `-z`?

```
var=true
something && var=false
# later
if $var && ...
```
Also, inside `[[ ]]` brackets quotes aren't needed for these things. Not sure why you added them again.

Reply:
Yes, unreachable sounds like the wrong variable name.

You can just assign `1` or an empty string in my opinion. The use of `-z` for the sake of negation is also ok. But I'd also really like to avoid the quotes on the lhs within `[[ ]]` (and you can also remove them on the rhs if not necessary like here).

Reply:
> Is that the intention?

Not sure how correct is it but yes. The logic I followed was for any not successful curl_output keep the handling as is. but skip it if there is at least an reason.

 > Also I find that long variable not very readable

I changed it. hopefully in something more meaningful

> inside [[ ]] brackets quotes aren't needed for these things. Not sure why you added them again.

That's partly my mistake. @baierjan asked me to add them but you had come before him asking to remove them.
The thing is that I know that in general is good practice to always use the quotes. So is there any link with what is good practice here to follow?

Reply:
Regarding `[[` in bash: https://stackoverflow.com/questions/3427872/whats-the-difference-between-and-in-bash/3427931#3427931

Reply:
I would also avoid quotes on the lhs of `[[` and on the rhs if the rhs value is a literal string without spaces.